### ISSUE TITLE
Creating first superadmin user

### ISSUE THREAD

**Issue by danielrama30** [sap-labs-france/ev-dashboard#2568]
I can't find a way how to create the first superadmin user.
I got the dashboard and the server both setup and running, but I can't login with default users from the test build.

**Comment by LucasBrazi06** [comment]
You have to register to the super admin tenant and activate it in the DB.

**Comment by danielrama30** [comment]
And how do I register to the tenant?

**Comment by LucasBrazi06** [comment]
Here :)
<img width="1141" alt="2021-09-20_14-26-05" src="https://user-images.githubusercontent.com/25266945/134001996-a7d6f249-2c3b-47e8-a8cf-022eaec1b398.png">

**Comment by danielrama30** [comment]
ok I tried that but it does nothing when i click to register the user

**Comment by LucasBrazi06** [comment]
> ok I tried that but it does nothing when i click to register the user

Check the console, you should have the error.
Have you setup your captcha?

**Comment by danielrama30** [comment]
I don't get any errors in neither the dashboard or server consoles, but the dashboard console spam this warning everytime I start it up:
Error from chokidar (/home/test/Downloads/ev-dashboard-2.4.88/node_modules/codeceptjs/node_modules/.bin): Error: ENOSPC: System limit for number of file watchers reached, watch '/home/test/Downloads/ev-dashboard-2.4.88/node_modules/codeceptjs/node_modules/.bin'

Oh and to be more specific, when I click register after putting every information for the user it's like it doen't literally send any request from the browser

**Comment by LucasBrazi06** [comment]
First time I see that :(
Do you use Chrome?
What OS?

**Comment by danielrama30** [comment]
Ok found the problem, it was the recaptcha wrong type (v2 instead of v3) and firefox not showing the error

Now I registered the account but how I activate it from the db and give it the super admin role?

**Comment by LucasBrazi06** [comment]
> Ok found the problem, it was the recaptcha wrong type (v2 instead of v3) and firefox not showing the error
> Now I registered the account but how I activate it from the db and give it the super admin role?

The new user is already a  super admin and use a tool like Robot3T and check the collection named 'default.users'.
Change the 'status' to 'A'.

**Comment by danielrama30** [comment]
ok, thanks now it´s working.

**Comment by LucasBrazi06** [comment]
You welcome ! :)

**Comment by danielrama30** [comment]
Another question still about creating users. How I'm supposed to login to the added organizations? I can't login with the super admin account and creating a new one seems working but I can't see any new user in the DB

**Comment by LucasBrazi06** [comment]
> Another question still about creating users. How I'm supposed to login to the added organizations? I can't login with the super admin account and creating a new one seems working but I can't see any new user in the DB

You entered an email when you created the org, and you should have received an email. Click on the link and create a password and you'll be able to login.

**Comment by danielrama30** [comment]
I didn´t receive anything and the email in config.json is configured correctly

**Comment by LucasBrazi06** [comment]
You can check the logs if there is an error?

**Comment by danielrama30** [comment]
From the Logs of dashboard there are no errors

**Comment by LucasBrazi06** [comment]
We log the email sent, find it and you'll get the link to activate your account in the new org.

**Comment by danielrama30** [comment]
where are these logs? tried to find them in the server files but without success

**Comment by LucasBrazi06** [comment]
Check both the collections default.logs or the <Org ID>.logs

**Comment by danielrama30** [comment]
Should be this log? it's immediately after the log of created organization
Message:
Express HTTP Response - 1499 ms - 0.05 KB >> POST/200 '/client/api/TenantCreate': Execution must be < 1000 ms, got 1499 ms

Detalied message:
[
 {
  "error": "Error: Execution must be < 1000 ms, got 1499 ms\n    at ServerResponse.<anonymous> (/home/test/Downloads/ev-server-2.4.88/dist/webpack:/src/utils/Logging.ts:339:25)\n    at ServerResponse.emit (events.js:412:35)\n    at onFinish (_http_outgoing.js:792:10)\n    at callback (internal/streams/writable.js:513:21)\n    at afterWrite (internal/streams/writable.js:466:5)\n    at afterWriteTick (internal/streams/writable.js:453:10)\n    at processTicksAndRejections (internal/process/task_queues.js:81:21)"
 }
]

**Comment by LucasBrazi06** [comment]
In the org logs, check the action = EmailNotification:

<img width="1425" alt="2021-09-20_17-03-09" src="https://user-images.githubusercontent.com/25266945/134025638-c9c716b3-61e1-49d0-96f1-91cd8214d6a9.png">

**Comment by LucasBrazi06** [comment]
I just checked and it works on my side

**Comment by danielrama30** [comment]
tried, it's empty

**Comment by LucasBrazi06** [comment]
What collections did you check?

**Comment by danielrama30** [comment]
default.logs both from dashboard and mongodb compass
![Screenshot from 2021-09-20 17-27-30](https://user-images.githubusercontent.com/43497685/134029222-12f8fc1d-ffad-4e7b-b602-f6c4c1aa1661.png)

**Comment by LucasBrazi06** [comment]
The log is in the new org and I don't see any collections starting with an ID with logs at the end.

When you create a new org, it creates a sub-set of collections like this:
<img width="1864" alt="2021-09-20_17-27-13" src="https://user-images.githubusercontent.com/25266945/134032467-1a6047d6-65be-43d0-8185-1549fed50291.png">

So I assume that something went wrong when you created the org!?

**Comment by danielrama30** [comment]
I have not logs like that, and I can't see logs from org dashboard since I can't login there.

**Comment by LucasBrazi06** [comment]
Sorry wrong screenshot, this one:
<img width="332" alt="2021-09-20_17-46-53" src="https://user-images.githubusercontent.com/25266945/134036038-1419c720-e329-47ac-90e8-bc261a4dc4da.png">

**Comment by danielrama30** [comment]
found the log, there's an error sending the email, but i found the link which is "http://cristomaiale.localhost:80/verify-email?VerificationToken=384386cb80a6029caa3cdc535273760afbda9cb9e9eff87e2b4c3a3ad1dbed26&Email=####&ResetToken=9231f80f-8475-4cbc-9ee8-f9783b491d4c" but it doesn't work

**Comment by LucasBrazi06** [comment]
> found the log, there's an error sending the email, but i found the link which is "http://cristomaiale.localhost:80/verify-email?VerificationToken=384386cb80a6029caa3cdc535273760afbda9cb9e9eff87e2b4c3a3ad1dbed26&Email=####&ResetToken=9231f80f-8475-4cbc-9ee8-f9783b491d4c" but it doesn't work

And what is the log now that you used the link? :-)

**Comment by danielrama30** [comment]
ok I tried changing port to 45000 and it takes me to a page where i should enter an email but i can't write into it

**Comment by danielrama30** [comment]
this is the detailed message in the next log:
{
    "_id" : ObjectId("6149a9c30c1a4e724dbe2093"),
    "userID" : null,
    "actionOnUserID" : null,
    "level" : "D",
    "chargingStationID" : null,
    "companyID" : null,
    "siteID" : null,
    "siteAreaID" : null,
    "source" : "Central Server",
    "host" : "test",
    "process" : "master",
    "type" : "S",
    "timestamp" : ISODate("2021-09-21T09:45:39.723Z"),
    "module" : "Logging",
    "method" : "logExpressRequest",
    "action" : "HttpRequest",
    "message" : "Express HTTP Request << GET '/verify-email?VerificationToken=adae32cc788f499dc364ea2bf4b14808c53a9922aa809572fdc5256521fff8b4&Email='",
    "detailedMessages" : "[\n {\n  \"url\": \"/verify-email?VerificationToken=adae32cc788f499dc364ea2bf4b14808c53a9922aa809572fdc5256521fff8b4&Email=####\",\n  \"method\": \"GET\",\n  \"query\": {\n   \"VerificationToken\": \"adae32cc788f499dc364ea2bf4b14808c53a9922aa809572fdc5256521fff8b4\",\n   \"Email\": \"####\"\n  },\n  \"body\": {},\n  \"locale\": \"en_US\",\n  \"xhr\": false,\n  \"ip\": \"127.0.0.1\",\n  \"ips\": [],\n  \"httpVersion\": \"1.1\",\n  \"headers\": {\n   \"host\": \"con9.localhost\",\n   \"user-agent\": \"Mozilla/5.0 (X11; Linux x86_64; rv:78.0) Gecko/20100101 Firefox/78.0\",\n   \"accept\": \"text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8\",\n   \"accept-language\": \"en-GB,en;q=0.5\",\n   \"accept-encoding\": \"gzip, deflate\",\n   \"dnt\": \"1\",\n   \"connection\": \"keep-alive\",\n   \"upgrade-insecure-requests\": \"1\"\n  }\n }\n]"
}

**Comment by danielrama30** [comment]
tried to use reset password and same, can see log with link, but when I use it it takes me to a blank page with the link "subomain.localhost:45000"

**Comment by danielrama30** [comment]
> > ok I tried changing port to 45000 and it takes me to a page where i should enter an email but i can't write into it
> 
> Normally, for a new org, you should get into the password creation.

I got that, but how I can resolve that? It seems to be the link wrong or something gets redirected

**Comment by LucasBrazi06** [comment]
Check the internal logs (default.logs and <tenantID>.logs)

**Comment by emry78** [comment]
> > ok I tried that but it does nothing when i click to register the user
> 
> Check the console, you should have the error. Have you setup your captcha?

Hi may i ask something. When i try to register user. At user password. I key in the password, follow the instruction 8 charater, pne lowercase and other. still cannot register. Need your advice. Thanks

![Screenshot 2022-04-18 at 11 06 02 AM](https://user-images.githubusercontent.com/49600677/163748887-303ebfa0-77d4-46a9-9446-b77b29ed5955.png)

**Comment by LucasBrazi06** [comment]
Try again? :)

**Comment by emry78** [comment]
> Try again? :)

ok done again. when i try to register. no response mean not go to page register. Any setting need to solve. pls advice.

**Comment by LucasBrazi06** [comment]
I don't get what you wrote.
Can you rephrase?

**Comment by emry78** [comment]
> 

https://user-images.githubusercontent.com/49600677/164575944-9107a131-9cde-4723-8a68-64a25232dccf.mov

you can see the short video this reply. When click button registers no response and action.
pls advice Sir. Thanks

**Comment by LucasBrazi06** [comment]
Check the Chrome console if you see an error.

**Comment by emry78** [comment]
> 

ok. It new for me. But I will go and read what the error occurring.
![Screenshotfrom2022-04-2215-48-54](https://user-images.githubusercontent.com/49600677/164644206-cbb00b60-649d-409f-a33a-26a23a2c0a42.png)
